cuda_bindings/cuda/bindings/cycufile.pxd

```diff
@@ -156,53 +167,61 @@ cdef extern from '<cufile.h>':
         CUFILE_PARAM_SKIP_TOPOLOGY_DETECTION
         CUFILE_PARAM_STREAM_MEMOPS_BYPASS
 
+cdef extern from '<cufile.h>':
     ctypedef enum CUFileStringConfigParameter_t:
         CUFILE_PARAM_LOGGING_LEVEL
         CUFILE_PARAM_ENV_LOGFILE_PATH
         CUFILE_PARAM_LOG_DIR
 
+cdef extern from '<cufile.h>':
     ctypedef enum CUFileArrayConfigParameter_t:
         CUFILE_PARAM_POSIX_POOL_SLAB_SIZE_KB
         CUFILE_PARAM_POSIX_POOL_SLAB_COUNT
 
     # types
-    ctypedef void* CUfileHandle_t 'CUfileHandle_t'
-    ctypedef void* CUfileBatchHandle_t 'CUfileBatchHandle_t'
+ctypedef void* CUfileHandle_t 'CUfileHandle_t'
+ctypedef void* CUfileBatchHandle_t 'CUfileBatchHandle_t'
+cdef extern from '<cufile.h>':
     ctypedef struct CUfileError_t 'CUfileError_t':
         CUfileOpError err
         CUresult cu_err
-    cdef struct _anon_pod0 '_anon_pod0':
-        unsigned int major_version
-        unsigned int minor_version
-        size_t poll_thresh_size
-        size_t max_direct_io_size
-        unsigned int dstatusflags
-        unsigned int dcontrolflags
+cdef struct _anon_pod0 '_anon_pod0':
```

Comment:
Was it intentional to move these from being externed to now no longer being externed? We should make sure this doesn't cause an ABI break if so.

Reply:
@kkraus14: The `cdef extern from '<cufile.h>':` means "this struct is external, use it from the header", not "make this an external symbol".  If you do that, Cython assumes the C struct exists in the header and won't generate one.  In this case, we need to generate one since the generator creates these as an implementation detail, and we need the structs in order to access their members.